### PR TITLE
Enable Automated Deployments

### PR DESCRIPTION
**Pull request by seanmalloy**
Use GitHub actions to automated container image build and deploy when
tags are pushed to the repo. Only automated container image builds are
enabled. Goreleaser is not enabled yet.

### CODE REVIEW

**Comment by seanmalloy** [comment]
I have no idea if the automated deploy to quay.io will work or not. My plan is to create a release after this is merged, and then figure out how to fix it if it fails.

File: .github/workflows/pipeline.yml [hunk 1]
```diff
@@ -42,3 +43,18 @@ jobs:
...
         verbose: true
+  deploy:
+    if: startsWith(github.ref, 'refs/tags/v')
+    needs: test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v2
+      - name: Push to quay.io
+        uses: docker/build-push-action@v1
```
Comment (vinny-sabatini) [inline .github/workflows/pipeline.yml:54]: It looks like there is a v2 of this action, should we start with that so we do not have to worry about upgrading later?
https://github.com/marketplace/actions/build-and-push-docker-images#upgrade-from-v1

**Review by vinny-sabatini — CHANGES_REQUESTED** [review]
Looks good, just one question around the action being used.

File: .github/workflows/pipeline.yml (hunk 1 above)
Comment (seanmalloy) [reply .github/workflows/pipeline.yml]: @vinny-sabatini fixed in commit 3fb3322565db73321246ef9c135b9cb42de00e71.

File: .github/workflows/pipeline.yml [hunk 2]
```diff
@@ -42,3 +43,31 @@ jobs:
...
+        with:
+          username: ${{ secrets.QUAY_USER }}
+          password: ${{ secrets.QUAY_TOKEN }}
+          registry: quay.io
+      - name: Build and push to quay.io
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          push: true
+          tags: ${{ steps.prep.outputs.tags }}
```
Comment (seanmalloy) [inline .github/workflows/pipeline.yml:73]: Probably won't update the `latest` tag in quay.io, but I'm ok with that for now. I just want to see if it works or not. A future PR can update the pipeline to also push the `latest` tag.

Comment (sanbornick) [reply .github/workflows/pipeline.yml:73]: I don't think so either, but fair.

**Review by sanbornick — APPROVED** [review]

**Comment by sanbornick** [comment]
`TAGS="${DOCKER_IMAGE}:${VERSION},${DOCKER_IMAGE}:latest"`

Can try that after if there isn't some magic on that particular tag.

**Pull request merged**